File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        
        visited = {}
        
        for c in s:
            if c not in visited:
                visited[c] = 1
                continue
                
            visited[c] += 1
        
        
        for c in t:
            if c not in visited:
                return False
            
            visited[c] -= 1
            
            if visited[c] < 0:
                return False
        
        
        return True

File: 0242-valid-anagram/0242-valid-anagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
    if (s.length != t.length){
        return false
    }
    obj = {}
    
    for (let i = 0;i<s.length;i++){
        if (obj[s[i]]){
            obj[s[i]] += 1
            continue
        }
        obj[s[i]] = 1
    }
    
    console.log(obj)
    for (let i = 0;i<t.length;i++){
        if (!obj[t[i]]){
            return false
        }
        obj[t[i]] -= 1
    }
    
    for (let k in obj){
        if (obj[k] != 0) return false
    }
    
    
    
    return true
};